pkg/scheduler/framework/uniquename/uniquename.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// package uniquename features some utilities that are used to generate unique names in use
+// by the scheduler.
+package uniquename
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+const (
+	uuidLength = 6
+)
+
+// minInt returns the smaller one of two integers.
+func minInt(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// ClusterResourceBindingUniqueName returns a unique name for a cluster resource binding in the
+// format of DNS subdomain names (RFC 1123).
+//
+// The name is generated using the following format:
+// * [CRP-NAME] - [TARGET-CLUSTER-NAME] - [RANDOM-SUFFIX]
```

Comment:
This would cause name conflicts should the scheduler needs to create bindings (of different states) for the same cluster -> actually this is related to some of the finer points (and some of the other comments in this PR) on which I feel some clarification would be most helpful. I will schedule a sync offline to further discuss this.